kernel/userspace.c

```diff
@@ -94,6 +96,144 @@ struct perm_ctx {
 	struct k_thread *parent;
 };
 
+#ifdef CONFIG_DYNAMIC_OBJECTS
+#include <tommyhashlin.h>
+
+extern struct _k_object *_k_object_gperf_find(void *obj);
+extern void _k_object_gperf_wordlist_foreach(_wordlist_cb_func_t func,
+					     void *context);
+
+tommy_hashlin dyn_objects;
+int dyn_objects_init;
+
+static int init_dyn_objects(struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	tommy_hashlin_init(&dyn_objects);
+	dyn_objects_init = 1;
+	return 0;
+}
+
+SYS_INIT(init_dyn_objects, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_OBJECTS);
+
+struct dyn_obj {
+	struct _k_object kobj;
+	tommy_node node;
+	u8_t data[];
+};
+
+static size_t obj_size_get(enum k_objects otype)
```

Comment:
This is another place that need to be changed when adding a new kernel object type. Can we get this auto generated as well? We already have the struct names/k_object references in gen_kobject_list.py

Reply:
once your generation patch lands I can make this change.